packages/core/src/physics/shape/ColliderShape.ts

```diff
@@ -0,0 +1,96 @@
+import { IColliderShape } from "@oasis-engine/design";
+import { PhysicsMaterial } from "../PhysicsMaterial";
+import { Quaternion, Vector3 } from "@oasis-engine/math";
+import { ignoreClone } from "../../clone/CloneManager";
+import { Collider } from "../Collider";
+
+/**
+ * Abstract class for collider shapes.
+ */
+export abstract class ColliderShape {
+  static idGenerator: number = 0;
```

Comment:
Should be hidden for developer by add `private` and named '_idGenerator'.

Reply:
done